### PR TITLE
Add enhancement change for disabling SC creation

### PR DESCRIPTION
**Pull request by gnufied**
Allow disabling storageclass creation

### CODE REVIEW

File: enhancements/storage/csi-driver-install.md [hunk 1]
```diff
@@ -176,6 +176,27 @@ Upgrade from 4.5 with Manila CSI driver (GA in 4.5) works as follows:
 In case there was no AWS / Manila CSI driver operator running during the update, CSO installs the corresponding operator
 as during installation, so user has a CSI driver running after update.
 
+### Disabling creation of storageclass
+Starting with OCP 4.12, we are allowing users to disable creation of StorageClass in case they need CSI driver but don't
+necessarily need dynamic provisioning in the cluster.
+
+An user can edit and set `disableStorageClass` field to `true` in `ClusterCSIDriver` object of that driver to disable
```
Comment (dobsonj) [inline enhancements/storage/csi-driver-install.md:183]: Tiny nit: should be `A user`, but +1 to these updates in general.

File: enhancements/storage/csi-driver-install.md [hunk 2]
```diff
@@ -176,6 +176,30 @@ Upgrade from 4.5 with Manila CSI driver (GA in 4.5) works as follows:
...
+```golang
+type ClusterCSIDriverSpec struct {
+    OperatorSpec `json:",inline"`
+    // DisableStorageClass determines if creation of storageclass for this driver
+    // should be skipped. By default Openshift creates storageclass for every CSI
+    // driver it installs via CSI operator. Setting this value to true will skip
+    // automatic creation of storageclass for driver being installed.
+    // Defaults to false.
+    // +kubebuilder:default=false
+    DisableStorageClass bool `json:"disableStorageClass,omitempty"`
```
Comment (jsafrane) [inline enhancements/storage/csi-driver-install.md:196]: Not sure if `DisableStorageClassCreation` is better.

File: enhancements/storage/csi-driver-install.md [hunk 3]
```diff
@@ -176,6 +176,30 @@ Upgrade from 4.5 with Manila CSI driver (GA in 4.5) works as follows:
 In case there was no AWS / Manila CSI driver operator running during the update, CSO installs the corresponding operator
 as during installation, so user has a CSI driver running after update.
 
+### Disabling creation of storageclass
+Starting with OCP 4.12, we are allowing users to disable creation of StorageClass in case they need CSI driver but don't
+necessarily need dynamic provisioning in the cluster.
```
Comment (gcharot) [inline enhancements/storage/csi-driver-install.md:181]: I think we should also add the use case where admins don't like how we configure the storage class and want to use their own instead.

File: enhancements/storage/csi-driver-install.md (hunk 2 above)
Comment (gcharot) [reply enhancements/storage/csi-driver-install.md:196]: +1 I think this is more intuitive

Comment (gnufied) [reply enhancements/storage/csi-driver-install.md:196]: As a user editable option - the longer the field name is,  the higher the chance is that they will make a typo and we will end up with a support case that it does not work as expected.

File: enhancements/storage/csi-driver-install.md (hunk 3 above)
Comment (gnufied) [reply enhancements/storage/csi-driver-install.md:181]: added

**Pull request merged**